### ISSUE TITLE
use validation set in whole game

### ISSUE THREAD

**Issue by topepo** [bmreiniger/aml4td-computing-python#3]
It looks like this [can be done](https://stackoverflow.com/a/31949393)

**Comment by bmreiniger** [comment]
It can, but requires concatenating the validation data with the training data, then passing an indicator array. I always found that a bit finicky: what if you don't concatenate right, or get the indicators wrong?  Here I guess we still have `X_train_val` around, and could extract and use the indices of `X_train` and `X_val`.

I can show that in a PR, but I think cv-ing inside the training set is more straightforward, and makes the inter-model comparisons on the validation set more honest anyway.
